php/classes/DB.php

```diff
@@ -0,0 +1,108 @@
+<?php
+
+require_once __DIR__ . '/../inc/configDB.php';
+
+class DB {
+  // (A) CONSTRUCTOR - CONNECT TO DATABASE
+  private $pdo = null;
+  private $stmt = null;
+  public $msg = "";
+  function __construct () {
+    try {
+      $this->pdo = new PDO(
+        "mysql:host=".DB_HOST.";dbname=".DB_NAME.";charset=".DB_CHARSET, 
+        DB_USER, DB_PASSWORD, [
+          PDO::ATTR_ERRMODE => PDO::ERRMODE_EXCEPTION,
+          PDO::ATTR_DEFAULT_FETCH_MODE => PDO::FETCH_ASSOC
+        ]
+      );
+    } catch (Exception $ex) { die($ex->getMessage()); }
+  }
+
+  // (B) DESTRUCTOR - CLOSE DATABASE CONNECTION
+  function __destruct () {
+    if ($this->stmt!==null) { $this->stmt = null; }
+    if ($this->pdo!==null) { $this->pdo = null; }
+  }
+
+  function queryFetchAll ($sql) {
```

Comment:
I think these guys (`queryFetchAll` and  `queryFetch `) are the only ones being generic enough in a class that is simply called `DB`. In order to respect the [single responsibility principle](https://en.wikipedia.org/wiki/Single-responsibility_principle), you should probably  keep this as a generic class `DB` and create a specific service `ReceiptService` class containing all the functions that are specific to the Receipts like `getFirstReceiptToCheck` and the others, using an instance of this generic class internally.